### DISCUSSION TITLE
MS SQL Server - Evaluate SQL expression force injects SELECT into statement

### DISCUSSION THREAD

**Discussion by kstrophus** [dbeaver/dbeaver#11789] [Q&A]
#### System information: 
- Operating system (distribution) and version: Windows 10
- DBeaver version: 7.3.5
- Additional extensions: None

#### Connection specification:
- Database name and version: MSSQL 2012
- Driver name: Microsoft JDBC Driver for SQL Server
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
Any attempt to evaluate a SQL statement fails because the word "SELECT" is injected into the script, whether already an select, update, etc.. However, executing the SQL itself succeeds. I need the evaluate while doing complex UPDATE queries with joins to check my syntax quickly as I go. This is very annoying.

#### Steps to reproduce, if exist:
- Connect to a MS SQL server.
- Open a SQL editor.
- Enter any SELECT or UPDATE SQL command, eg. something simple: ```SELECT @@VERSION```
- "SQL Editor" menu "Evaluate SQL Expression"
- Notice in the small pane in DBeaver that previews the expression: ```SELECT SELECT @@VERSION```

#### Include any warning/errors/backtraces from the logs
org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [156] [S0001]: Incorrect syntax near the keyword 'select'.

**Comment by uslss** [comment]
thanks for the report

**Comment by LonwoLonwo** [comment]
Hello @kstrophus 
"Evaluate SQL Expression" - just small helpful function to execute part of the query.
you can read about it here: https://github.com/dbeaver/dbeaver/wiki/SQL-Execution#sql-expression-evaluation
This is a pattern, that always adds SELECT at the query beginning (and depending on the database "FROM DUAL").

"INSERT INTO datetable(a) values (current_timestamp())" - as example, you can use "Evaluate SQL Expression" for "current_timestamp()".

In you case - use "Evaluate SQL Expression" for "@@VERSION" or ctrl+Enter for "SELECT @@VERSION"

**Reply by kstrophus** [reply]
That menu wording is very confusing. So how do I check an SQL expression without triggering an actual execution? I want to be able to evaluate the SQL expression syntax as I write a more complex update query (or any query, for that matter).

**Reply by p91paul** [reply]
> That menu wording is very confusing. So how do I check an SQL expression without triggering an actual execution? I want to be able to evaluate the SQL expression syntax as I write a more complex update query (or any query, for that matter).

I think you are confusing the meaning of "evaluate" with "validate". Evaluate means "get the value of", and here is used correctly.

You are looking for a "**validate** expression" button, which does not exist (and I believe would be quite difficult to implement).

**Reply by LonwoLonwo** [reply]
Sorry @kstrophus , but @p91paul is right. It's quite difficult to implement this feature, that you suggested. 
SQL expression can not be executed on the client side, it may include calling functions and use the variables that only exist on the server

**Comment by LonwoLonwo** [comment]
The ticket moves into the discussion
